index.js

```diff
@@ -216,14 +218,25 @@ function request(method, url, options, callback) {
 
   var responded = false;
 
-  var req = protocols[url.protocol.replace(/\:$/, '')].request({
+  var params = {
     host: url.hostname,
     port: url.port,
     path: url.path,
     method: method,
     headers: rawHeaders,
     agent: agent
-  }, function (res) {
+  }
+
+  if (options.proxy === null || options.proxy === undefined || options.proxy === '') {
+    options.proxy = 'http_proxy' in env ? env.http_proxy : false;
```

Comment:
Well, it's very common in the UNIX world and also e.g. the Java VM or standard command line tools like wget use it. If you [google for it](https://www.google.de/search?q=http_proxy) you see it many times. Why not give your fellow developers the benefit to not implement this every time on their own?